File: src/main/java/me/captainpotatoaim/myplugin/random_commands/protect_area/Area.java
package me.captainpotatoaim.myplugin.random_commands.protect_area;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

import java.util.*;

public record Area(SerializableLocation start, SerializableLocation end) {
    boolean contains(SerializableLocation location) {
        int startX = start.getX();
        int startY = start.getY();
        int startZ = start.getZ();

        int endX = end.getX();
        int endY = end.getY();
        int endZ = end.getZ();

        int x = location.getX();
        int y = location.getY();
        int z = location.getZ();

        return ((startX <= x && x <= endX) || (startX >= x && x >= endX))
                && ((startY <= y && y <= endY) || (startY >= y && y >= endY))
                && ((startZ <= z && z <= endZ) || (startZ >= z && z >= endZ));
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Area area = (Area) o;

        if (!Objects.equals(start, area.start)) return false;
        return Objects.equals(end, area.end);
    }

    @Contract(pure = true)
    @Override
    public @NotNull String toString() {
        return "Area{" +
                "start=" + start +
                ", end=" + end +
                '}';
    }
}

File: src/main/java/com/github/anviks/vixplugin/random_commands/protect_area/Area.kt
package com.github.anviks.vixplugin.random_commands.protect_area

import kotlinx.serialization.Serializable

@Serializable
data class Area(private val start: SerializableLocation, private val end: SerializableLocation) {
    fun contains(location: SerializableLocation): Boolean {
        val startX = start.x
        val startY = start.y
        val startZ = start.z

        val endX = end.x
        val endY = end.y
        val endZ = end.z

        val x = location.x
        val y = location.y
        val z = location.z

        return ((startX <= x && x <= endX) || (startX >= x && x >= endX))
                && ((startY <= y && y <= endY) || (startY >= y && y >= endY))
                && ((startZ <= z && z <= endZ) || (startZ >= z && z >= endZ))
    }
}
